src/odb/include/odb/gdsin.h

```diff
@@ -0,0 +1,160 @@
+#pragma once
+
+#include <cstdint>
+#include <fstream>
+#include <iostream>
+#include <vector>
+#include <endian.h>
+
+#include "odb/db.h"
+#include "odb/gdsUtil.h"
+
+namespace odb {
+
+class dbGDSLib;
+class dbGDSElement;
+class dbGDSBoundary;
+class dbGDSPath;
+class dbGDSSRef;
+class dbGDSStructure;
+
+class GDSReader
+{
+ public: 
+  /**
+   * Constructor for GDSReader
+   * No operations are performed in the constructor
+   */
+  GDSReader();
+
+  /**
+   * Destructor
+   * 
+   * Does not free the dbGDSLib objects, as they are owned by the database
+   */
+  ~GDSReader();
+
+  /**
+   * Reads a GDS file and returns a dbGDSLib object
+   * 
+   * @param filename The path to the GDS file
+   * @param db The database to store the GDS data
+   * @return A dbGDSLib object containing the GDS data
+   * @throws std::runtime_error if the file cannot be opened, or if the GDS is corrupted
+   */
+  dbGDSLib* read_gds(const std::string& filename, dbDatabase* db);
+
+ private:
+  /** Current filestream */
+  std::ifstream _file;
```

Comment:
data members should come after methods (https://google.github.io/styleguide/cppguide.html#Declaration_Order)